director/sort.go

```diff
@@ -144,21 +148,41 @@ func checkOverrides(addr net.IP) (coordinate *Coordinate) {
 	return nil
 }
 
-func getLatLong(addr netip.Addr) (lat float64, long float64, err error) {
+func getLatLong(ctx context.Context, addr netip.Addr) (lat float64, long float64, err error) {
 	ip := net.IP(addr.AsSlice())
 	override := checkOverrides(ip)
 	if override != nil {
 		log.Infof("Overriding Geolocation of detected IP (%s) to lat:long %f:%f based on configured overrides", ip.String(), (override.Lat), override.Long)
 		return override.Lat, override.Long, nil
 	}
 
+	network, ok := utils.ApplyIPMask(addr.String())
+	if !ok {
+		log.Warningf("Failed to apply IP mask to address %s", ip.String())
+	}
+
+	project, ok := ctx.Value(ProjectContextKey{}).(string)
+	if !ok {
+		log.Warningf("Failed to get project from context")
+		project = "unknown"
+	}
 	reader := maxMindReader.Load()
 	if reader == nil {
 		err = errors.New("No GeoIP database is available")
+		metrics.PelicanDirectorGeoIPErrors.With(prometheus.Labels{
+			"network": network,
+			"source":  "server",
+			"project": project,
```

Comment:
You already know the question I'm going to ask about this 🤓 How can we make sure a malicious client doesn't blow up cardinality by setting bogus projects in the user agent? This might be harder to solve, since the Director doesn't have a list of known projects ahead of time (or at any point). Is there a way to set a cardinality limit so that it's safe to pass user-defined input through to Prometheus?